src/Relude/Extra/Tuple.hs

```diff
@@ -60,3 +64,47 @@ mapToSnd f a = (a, f a)
 mapBoth :: (a -> b) -> (a, a) -> (b, b)
 mapBoth f (a1, a2) = (f a1, f a2)
 {-# INLINE mapBoth #-}
+
+{- | Apply a function that returns a value inside of a functor,
+with the output in the first slot, the input in the second,
+and the entire tuple inside the functor.
+
+
+A dual to 'traverseToSnd'
+
+>>> traverseToFst (Just . (+1)) 10
+Just (11,10)
+>>> traverseToFst (const Nothing) 10
+Nothing
+-}
+traverseToFst :: Functor t => (a -> t b) -> a -> t (b, a)
+traverseToFst f a = (,a) <$> f a
+{-# INLINE traverseToFst #-}
+
+{- | Apply a function that returns a value inside of a functor,
+with the output in the second slot, the input in the first,
+and the entire tuple inside the functor.
+
+A dual to 'traverseToFst'.
+
```

Comment:
> functions that have type like `FilePath -> IO (FilePath, String)` seems valuable, I think this text also can be added to the documentation 📚

I can't think of good wording for this. Any advice?

Reply:
Hmm.. I see that the existing documentation is already good enough. And examples are nice. Probably don't need to tell more here.